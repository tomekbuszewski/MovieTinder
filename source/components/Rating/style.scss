@import '../App/globals';

.Rating {
  display: block;
  margin: 10px auto;
  max-width: 80%;
  max-height: 5rem;

  &__Wrapper {
    position: relative;
  }

  &__Text {
    position: absolute;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
  }

  &__Circle {
    fill: none;
    stroke-width: 2.8;
    stroke-linecap: round;
    animation-name: progress;
    animation-duration: 500ms;
    animation-fill-mode: forwards;

    &--Red { stroke: map_get($colors, red); }
    &--Yellow { stroke: map_get($colors, yellow); }
    &--Green { stroke: map_get($colors, green); }
  }
}

@keyframes progress {
  0% {
    stroke-dasharray: 0 100;
  }
}