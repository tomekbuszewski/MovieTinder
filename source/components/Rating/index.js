import React from 'react';
import ClassNameBuilder from 'bem-classname-builder';

import style from './style.scss';

const bem = new ClassNameBuilder(style, 'Rating');

/**
 * Rating svg animated circle. SVG found somewhere on the 'net long
 * time ago, can't seem to find the source.
 * @param {number} number - number 0 - 100
 * @returns {*} - rendered component
 * @constructor
 */
const Rating = ({ number }) => {
  let color = 'Green';

  if (number < 40) { color = 'Red'; }
  if (number >= 40 && number < 75) { color = 'Yellow'; }

  return <div className={bem.get({e:'Wrapper'})}>
    <span className={bem.get({e:'Text'})}>{number}%</span>
    <svg viewBox="0 0 36 36" className={bem.get()}>
      <path className={`${bem.get({e:'Circle'})} ${bem.get({e:'Circle', m: color})}`}
            strokeDasharray={`${number}, 100`}
            d="M18 2.0845
        a 15.9155 15.9155 0 0 1 0 31.831
        a 15.9155 15.9155 0 0 1 0 -31.831"
      />
    </svg>
  </div>
};

export default Rating;