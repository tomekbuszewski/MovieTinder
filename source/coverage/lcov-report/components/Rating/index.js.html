<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for components/Rating/index.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> / <a href="index.html">components/Rating</a> index.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">75% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>6/8</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">66.67% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>4/6</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>1/1</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>6/6</span>
      </div>
    </div>
  </div>
  <div class='status-line medium'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">18x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">18x</span>
<span class="cline-any cline-yes">18x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">18x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react';
import ClassNameBuilder from 'bem-classname-builder';
&nbsp;
import style from './style.scss';
&nbsp;
const bem = new ClassNameBuilder(style, 'Rating');
&nbsp;
/**
 * Rating svg animated circle. SVG found somewhere on the 'net long
 * time ago, can't seem to find the source.
 * @param {number} number - number 0 - 100
 * @returns {*} - rendered component
 * @constructor
 */
const Rating = ({ number }) =&gt; {
  let color = 'Green';
&nbsp;
  <span class="missing-if-branch" title="if path not taken" >I</span>if (number &lt; 40) { <span class="cstat-no" title="statement not covered" >color = 'Red'; </span>}
  <span class="missing-if-branch" title="if path not taken" >I</span>if (number &gt;= 40 &amp;&amp; number &lt; 75) { <span class="cstat-no" title="statement not covered" >color = 'Yellow'; </span>}
&nbsp;
  return &lt;div className={bem.get({e:'Wrapper'})}&gt;
    &lt;span className={bem.get({e:'Text'})}&gt;{number}%&lt;/span&gt;
    &lt;svg viewBox="0 0 36 36" className={bem.get()}&gt;
      &lt;path className={`${bem.get({e:'Circle'})} ${bem.get({e:'Circle', m: color})}`}
            strokeDasharray={`${number}, 100`}
            d="M18 2.0845
        a 15.9155 15.9155 0 0 1 0 31.831
        a 15.9155 15.9155 0 0 1 0 -31.831"
      /&gt;
    &lt;/svg&gt;
  &lt;/div&gt;
};
&nbsp;
export default Rating;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sun Jan 07 2018 14:56:53 GMT+0100 (CET)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
</body>
</html>
