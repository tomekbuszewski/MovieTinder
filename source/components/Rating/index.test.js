import React from 'react';
import renderer from 'react-test-renderer';

import Rating from './index';

describe ('Rating tests', () => {
  it ('should match snapshot', () => {
    const tree = renderer.create(<Rating number={90} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
