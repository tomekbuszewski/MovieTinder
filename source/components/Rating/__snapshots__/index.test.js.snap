// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rating tests should match snapshot 1`] = `
<div
  className=""
>
  <span
    className=""
  >
    90
    %
  </span>
  <svg
    className=""
    viewBox="0 0 36 36"
  >
    <path
      className=" "
      d="M18 2.0845 a 15.9155 15.9155 0 0 1 0 31.831 a 15.9155 15.9155 0 0 1 0 -31.831"
      strokeDasharray="90, 100"
    />
  </svg>
</div>
`;
